velox/exec/Driver.cpp

```diff
@@ -331,13 +331,28 @@ void Driver::pushdownFilters(int operatorIndex) {
 }
 
 RowVectorPtr Driver::next(std::shared_ptr<BlockingState>& blockingState) {
-  enqueueInternal();
   auto self = shared_from_this();
   facebook::velox::process::ScopedThreadDebugInfo scopedInfo(
       self->driverCtx()->threadDebugInfo);
   ScopedDriverThreadContext scopedDriverThreadContext(*self->driverCtx());
   RowVectorPtr result;
-  auto stop = runInternal(self, blockingState, result);
+
+  StopReason stop;
+
+  // Spin wait when task is paused.
```

Comment:
Instead of spin wait, can we return a future from Task::next and have the caller come back when future completes?

Reply:
That's right, I thought it should be feasible to return a future to caller somehow. But that may still require for some designs against the current common API `runInternal` and need to consider together with `BlockingState` which already have a future inside it. So I'd suggest doing the refactor in another PR so we can be flexible to design and discuss. Does that work for you? 

Also the code is currently only used when multiple independent single-thread tasks sharing the same shared arbitrator. So it just affects some use cases of the functionality.

Reply:
@zhztheplayer Task::next already allows for returning a Future and Driver::next supports that by producing a BlockingState. Let's figure out how to use these for this scenario. We cannot merge "spin wait" solution.

Reply:
@zhztheplayer Would you share a draft of a proper solution? I'm ware of landing spin-wait.

Reply:
OK I understand. I am on weekend currently and will do with some initial attempts later. Thanks.